src/Service.Host/client/src/components/SerialNumber.js

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import EditIcon from '@material-ui/icons/Edit';
+import Clear from '@material-ui/icons/Clear';
+import Done from '@material-ui/icons/Done';
+import { Button, TableRow, TableCell } from '@material-ui/core';
+import { InputField } from '@linn-it/linn-form-components-library';
+import moment from 'moment';
+import { getSelfHref } from '../helpers/utilities';
+
+function SerialNumber({ serialNumber, item, updateSernosNote, addSernosNote }) {
+    const [editing, setEditing] = useState(false);
+    const [sernosNote, setSernosNote] = useState({});
+    const [prevSernosNote, setPrevSernosNote] = useState({});
+
+    useEffect(() => {
+        if (item !== prevSernosNote) {
+            if (item !== null) {
+                setSernosNote(item);
+                setPrevSernosNote(item);
+            } else {
```

Comment:
what is the difference between serialNumber and item?

Reply:
This was to try and avoid having terrible names for the local component state. I used item as generally in our other components we pass the object of interest in as 'item'. Was trying to avoid having a variable name like sernosNoteLocal for the component state.